src/lsp/cobol_parser/grammar.mly

```diff
@@ -2894,15 +2913,33 @@ let accept_misc :=
  | EXCEPTION; STATUS;      {AcceptExceptionStatus} (* MF *)
 
 let accept_clause :=
- | ~ = at_position;                    <AcceptAt>
- | FROM; CRT;                          {AcceptFromCRT}   (* MF *)
- | MODE; IS?; BLOCK;                   {AcceptModeBlock} (* MF *)
- | WITH; ~ = loc(accept_with_clause)+; <AcceptWith>      (* MF *)
+ | ~ = at_position;                                  <AcceptAt>
+ | FROM; CRT;                                        {AcceptFromCRT}   (* MF *)
+ | MODE; IS?; BLOCK;                                 {AcceptModeBlock} (* MF *)
+ | WITH?; ~ = nell(loc(accept_with_clause)); %prec lowest <AcceptWith> (* MF *)
+   (* Note: GnuCOBOL makes "WITH" optional *)
 
-let accept_with_clause :=
- | UPDATE;                      {AcceptUpdate}
+let accept_with_clause [@context accept_stmt] := (* MF *)
  | ~ = screen_attribute_clause; <AcceptAttribute>
```

Comment:
If I don't add the `accept_stmt` context here, some context-sensitive keywords in this clause are not recognized (eg: AUTO)